nomad/resource_variable.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+const (
+	// maxPathLength is defined by Nomad here:
+	// https://github.com/hashicorp/nomad/blob/c7c8bd4ce2fcd9f2cd13278043b7baf1f8dd4f47/nomad/structs/variables.go#LL158
+	maxPathLength = 128
+)
+
+var (
+	validVariablePath = regexp.MustCompile("^[a-zA-Z0-9-_~/]+$")
+)
+
+func resourceVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVariableWrite,
+		Update: resourceVariableWrite,
+		Delete: resourceVariableDelete,
+		Read:   resourceVariableRead,
+		Exists: resourceVariableExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Description:  "The path at which the variable items will be stored, must be between 1 and 128 characters in length, be URL safe, and not include '@' or '.' characters",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: pathValidation(),
+			},
+			"namespace": {
+				Description: "Variable namespace",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     api.DefaultNamespace,
+			},
+			"items": {
+				Description: "A map of strings to be added as items in the variable",
+				Type:        schema.TypeMap,
+				Required:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func resourceVariableWrite(d *schema.ResourceData, meta any) error {
+	client := meta.(ProviderConfig).client
+
+	variable := &api.Variable{
+		Namespace: d.Get("namespace").(string),
+		Path:      d.Get("path").(string),
+		Items:     make(map[string]string),
+	}
+
+	for name, value := range d.Get("items").(map[string]any) {
+		variable.Items[name] = value.(string)
+	}
+
+	log.Printf("[DEBUG] Upserting Variable %s@%s", variable.Path, variable.Namespace)
+	if _, _, err := client.Variables().Create(variable, nil); err != nil {
+		return fmt.Errorf("error creating Variable %s@%s: %s", variable.Path, variable.Namespace, err.Error())
```

Comment:
```suggestion
		return fmt.Errorf("error creating variable %s@%s: %s", variable.Path, variable.Namespace, err)
```

Just some minor nit-picking 😅